lib/phobos/actions/process_batch.rb

```diff
@@ -20,15 +20,31 @@ def initialize(listener:, batch:, listener_metadata:)
 
       def execute
         instrument('listener.process_batch', @metadata) do |_metadata|
-          @batch.messages.each do |message|
-            Phobos::Actions::ProcessMessage.new(
-              listener: @listener,
-              message: message,
-              listener_metadata: @listener_metadata
-            ).execute
+          if @listener.handler_class.method_defined?(:consume_batch)
```

Comment:
This is also a bit error prone way of inferring the control flow (I have a related comment below mentioning `batch_inline` as an explicit delivery option [#batch_inline])

The thing this idea could enable is to initialize `ProcessBatch` with delivery method = one of `batch` / `batch_inline`. This would change the signature of initialize, but you can always default this to `batch`. And btw please make them constants in this file for now, eg `const DELIVERY_BATCH = 'batch'` and `const DELIVERY_INLINE = 'inline_batch'`

Then you can be more clear here in this if statement say with confidence:

`if @delivery_option == DELIVERY_INLINE`